torch/optim/optimizer.py

```diff
@@ -255,9 +255,14 @@ def __init__(self, params: ParamsT, defaults: Dict[str, Any]) -> None:
         self._patch_step_function()
 
         if isinstance(params, torch.Tensor):
-            raise TypeError("params argument given to the optimizer should be "
-                            "an iterable of Tensors or dicts, but got " +
-                            torch.typename(params))
+            if self.__class__.__name__ == 'SparseAdam':
+                warnings.warn(("Passing in a raw Tensor is deprecated. In the future, "
+                               "this will raise an error. Please wrap your Tensor in "
+                               "an iterable instead."), UserWarning)
```

Comment:
nit
```suggestion
                               "an iterable instead."), FutureWarning)
```

Reply:
What's the difference here?

Reply:
`FutureWarning` is a warning message that warns you about deprecated features that will be removed in a future version of a package.

I don't think we are that consistent about using `FutureWarning`, but I suppose it makes sense to do this in case users want to explicitly suppresss/allow certain warnings